mock/src/components/REPLInput.tsx

```diff
@@ -1,47 +1,87 @@
 import "../styles/main.css";
 import { Dispatch, SetStateAction, useState } from "react";
 import { ControlledInput } from "./ControlledInput";
+import { HistoryElement } from "./REPL";
+import { populateCommandRegistry, commandRegistry } from "./Commands";
 
+/**
+ * Props interface for REPLInput component.
+ */
 interface REPLInputProps {
-  history: string[];
-  setHistory: Dispatch<SetStateAction<string[]>>;
+  /**
+   * List of history elements and its setter.
+   */
+  history: HistoryElement[];
+  setHistory: Dispatch<SetStateAction<HistoryElement[]>>;
+
+  /**
+   * Flag and setter indicating if page is in verbose mode or not.
+   */
   verbose: boolean;
   setVerbose: Dispatch<SetStateAction<boolean>>;
-  // TODO: Fill this with desired props... Maybe something to keep track of the submitted commands
+  //handleCommand: Function;
 }
-// You can use a custom interface or explicit fields or both! An alternative to the current function header might be:
-// REPLInput(history: string[], setHistory: Dispatch<SetStateAction<string[]>>)
+
+/**
+ * Component for handling user input and displaying the REPL interface. Does not display directly but
+ * passes information up to REPL about what the user provided.
+ * @param {REPLInputProps} props - Props for the REPLInput component.
+ * @returns {JSX.Element} The REPL input component.
+ */
 export function REPLInput(props: REPLInputProps) {
-  // Remember: let React manage state in your webapp.
-  // Manages the contents of the input box
   const [commandString, setCommandString] = useState<string>("");
   const [count, setCount] = useState<number>(0);
-  // TODO WITH TA : add a count state
 
-  // TODO WITH TA: build a handleSubmit function called in button onClick
-  // TODO: Once it increments, try to make it push commands... Note that you can use the `...` spread syntax to copy what was there before
-  // add to it with new commands.
+  //populate command map with base functions then call executefunction(command) from map
+  populateCommandRegistry();
+
   /**
-   * We suggest breaking down this component into smaller components, think about the individual pieces
-   * of the REPL and how they connect to each other...
+   * Handles user clicking the submit button, adds command and output to list of history elements.
    */
   const handleSubmit = () => {
-    setCount(count + 1);
     //console.log("Command string: " + commandString);
-    if (commandString === "mode"){
-      console.log("boolean evaluated true");
-      props.setVerbose(!props.verbose);
+    const args = commandString.split(/\s+/); // Split command string into arguments
+    const commandName = args.shift(); // Extract the command name (first index of list)
+
+    if (commandName) {
+      if (commandString === "mode") {
```

Comment:
Perhaps mode as a command would be optimally stored in the command registry with the other commands, and you could use some form of shared state to pass the current mode to the command function.